src/index.ts

```diff
@@ -53,11 +123,29 @@ export const getRequestBody = (
 
   requestBody['pd[tm]'] = payload.tm || 'pinterest-mc'
 
+  // match  event types to Pinterest's default
+
+  if (ecommerce) {
+    switch (eventType) {
+      case 'Product Added':
+        requestBody.event = 'addtocart'
+        break
+      case 'Order Completed':
+        requestBody.event = 'checkout'
+        break
+    }
+    // include all ecommerce parameters
+
+    const ecommerceData: any = mapEcommerceData(ecommerce) // not sure how to get read of this any!
```

Comment:
I'd start by adding a concrete return type to the `mapEcommerceData` function

Reply:
Thank you, added!